src/features/profile/components/UserProfile/index.tsx

```diff
@@ -0,0 +1,54 @@
+'use client';
+import { IconPen } from '@/assets/images/icons';
+import { Button, Icon } from '@/components/ui';
+import { IUser } from '@/types/definition';
+
+import Image from 'next/image';
+import { useState } from 'react';
+
+import { RenameUserForm } from './RenameUserForm';
+
+interface IUserProfileProps {
+  /**
+   * The id of the user.
+   */
+  userId: IUser['id'];
+  /**
+   * The name of the user.
+   */
+  name: IUser['name'];
+}
+
+export const UserProfile = ({ userId, name }: IUserProfileProps) => {
+  const [isRenameUserFormOpen, setIsRenameUserFormOpen] = useState(false);
+
+  /**
+   * Open the rename user form when the pen icon is clicked.
+   */
+  const handlePenIconClick = () => {
+    setIsRenameUserFormOpen(true);
+  };
+
+  return (
+    <div className="flex flex-col items-center gap-4 pt-6">
+      <Image src="/images/avatar.png" width={128} height={128} alt="avatar" priority />
```

Comment:
I didn't use shadcn's `Avatar` component, mainly because
- Didn't want to extend this task's scope to implement new UI component
- For now this is the only avatar, and couldn't figure out how it's different from normal img tag

Also, to set the size of this avatar, I **didn't** utilize tailwind utillity width like this ⬇️
```tsx
<div className="w-32 h-32 relative">
  <Image src="/images/avatar.png"  alt="avatar" fill priority />
</div>
``` 
Because of the 2 reasons ⬇️ 
- It's anyway fixed size, and use built-in `width` property seems to be simpler.
- Setting "fill" to `<Image/>` tag caused this caution ⬇️ , and it required to add `sizes` attribute.
 <img width="717" alt="Screenshot 2024-03-06 at 23 57 31" src="https://github.com/nishiki-tech/nishiki-frontend/assets/99339182/f005680c-3610-4fe3-986b-7d85a15e4ab9">

 Setting `sizes` attribute as NextJS suggest seemed to be making things unnecessarily complicated.

 Reference : https://nextjs.org/docs/pages/api-reference/components/image#sizes

Reply:
My conclusion is either:

1. Using the SVG component as it is without `next/image` here.
2. Converting the SVG image into PNG and using it with `next/image` here.

But, in either case, place the image file under the `src/assets/images/ui/` directory, delete the old `icon_person-circle.svg` file, and replace all of them with the new file Also, using the `<Icon>` component should be avoided.

Usually, commonly we can import an SVG file as an "image" and specify it to `<Image src" ~ here ~ ">`. However, as you might have researched, we cannot take this approach because, in our project, all SVG images are automatically transformed into a React Component by `@svgr/webpack` being set up for the entire project.

### Approach 1 (SVG):
I prefer this because it's simpler.
```tsx
// SVG
<PersonCircle className="size-32" />
```

### Approach 2 (PNG):
```tsx
// PNG
<Image src={PersonCircle} width="128" heigh="128" />
```

By the way, when you specify `<Image src" ~ this value ~ ">`, imported images are usually better than manual paths unless you need to dynamically define the `src` value. When you refactor or rename the image files, TypeScript can cause an error if you use imported images, while manual paths fail to display the images on the client side.